File: testdata/csharp/Calc/Calculator.cs
namespace Calc;

using System;

public partial class Calculator
{
    public int Total { get; set; }

    [Obsolete("demo")]
    public Calculator(int seed)
    {
        Total = seed;
    }

    public string Classify(
        int score,
        int threshold)
    {
        if (score > threshold)
        {
            return "high";
        }
        if (score < threshold)
        {
            return "low";
        }
        return "equal";
    }

    private static int Helper(int n)
    {
        return n * 2;
    }

    public class Inner
    {
        public void Ping()
        {
        }
    }
}

File: testdata/java/src/main/java/com/example/calc/Calculator.java
package com.example.calc;

public class Calculator {

    private final int base;

    public Calculator(int base) {
        this.base = base;
    }

    public int add(int a, int b) {
        return a + b + base;
    }

    public <T extends Comparable<T>> String classify(
            T value,
            T threshold) {
        if (value.compareTo(threshold) > 0) {
            return "high";
        } else if (value.compareTo(threshold) < 0) {
            return "low";
        }
        return "equal";
    }

    static class Helper {
        int helperOnly(int x) {
            return x * 2;
        }
    }
}
